src/config.cpp

```diff
@@ -511,7 +511,7 @@ void config::recursive_clear_value(config_key_type key)
 	map_erase_key(values_, key);
 
 	for(std::pair<const std::string, child_list>& p : children_) {
-		for(config* cfg : p.second) {
+		for(auto& cfg : p.second) {
```

Comment:
well sure, auto would copy the object which fails to complie since unique_ptr has no cctor.